app/lib/spam_check.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class SpamCheck
+  include Redisable
+  include ActionView::Helpers::TextHelper
+
+  LEVENSHTEIN_THRESHOLD = 10
+
+  def initialize(status)
+    @account = status.account
+    @status  = status
+  end
+
+  def skip?
+    already_flagged? || no_unsolicited_mentions?
```

Comment:
I think it should also skip the check if it's a reply to a toot that mentioned the user.